diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3bcb909a..24839fb9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,7 +140,6 @@ add_subdirectory(crv)
 add_subdirectory(spr)
 add_subdirectory(ree)
 add_subdirectory(sam)
-add_subdirectory(phasta)
 add_subdirectory(stk)
 add_subdirectory(dsp)
 add_subdirectory(omega_h)
diff --git a/apf/CMakeLists.txt b/apf/CMakeLists.txt
index fc5aaf34..c27e172e 100644
--- a/apf/CMakeLists.txt
+++ b/apf/CMakeLists.txt
@@ -80,6 +80,7 @@ set(HEADERS
   apfField.h
   apfFieldData.h
   apfNumberingClass.h
+  apfElement.h
 )
 
 # Add the apf library
diff --git a/apf/apfElement.cc b/apf/apfElement.cc
index 161de8df..d35bb8f3 100644
--- a/apf/apfElement.cc
+++ b/apf/apfElement.cc
@@ -121,9 +121,8 @@ void Element::getNodeData()
 
 void Element::getElementNodeData(NewArray<double>& d)
 {
-  d.allocated() ? d.resize(nen) : d.allocate(nen);
-  for (int i = 0; i < nen; i++)
-    d[i] = nodeData[i];
+  d.resize(nodeData.size());
+  std::copy(&nodeData[0], (&nodeData[nodeData.size()-1])+1, &d[0]);
 }
 
 }//namespace apf
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index d21dbb3a..6650b8a3 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -71,7 +71,6 @@ endif()
 
 if(ENABLE_SIMMETRIX)
   util_exe_func(convert convert.cc)
-  util_exe_func(ph_convert ../phasta/ph_convert.cc)
   if(SIMMODSUITE_SimAdvMeshing_FOUND)
     util_exe_func(generate generate.cc)
     if(SIM_DOT_VERSION VERSION_GREATER 12.0.171000)
@@ -111,24 +110,6 @@ util_exe_func(fixlayer fixlayer.cc)
 util_exe_func(fixDisconnected fixDisconnected.cc)
 util_exe_func(scale scale.cc)
 
-# Phasta utilities
-util_exe_func(chef ../phasta/chef.cc)
-util_exe_func(condense ../phasta/condense.cc)
-util_exe_func(threshold ../phasta/threshold.cc)
-util_exe_func(chefReadUrPrep ../phasta/readUrPrep.cc)
-if(NOT APPLE)
-  util_exe_func(chefStream ../phasta/chefStream.cc)
-  util_exe_func(adaptLvlSetLoop ../phasta/adaptLvlSet_loop.cc)
-endif()
-if(ENABLE_SIMMETRIX)
-  util_exe_func(cut_interface ../phasta/cut_interface.cc)
-  util_exe_func(migrate_interface ../phasta/migrate_interface.cc)
-endif()
-if(SIMMODSUITE_SimAdvMeshing_FOUND)
-  target_compile_definitions(chef PRIVATE -DHAVE_SIMADVMESHING)
-  target_compile_definitions(cut_interface PRIVATE -DHAVE_SIMADVMESHING)
-endif()
-
 #PUMIPic print partition utility
 util_exe_func(print_pumipic_partition print_pumipic_partition.cc)
 
@@ -181,7 +162,7 @@ test_exe_func(nedelecShapes nedelecShapes.cc)
 test_exe_func(L2Shapes L2Shapes.cc)
 test_exe_func(H1Shapes H1Shapes.cc)
 test_exe_func(poisson poisson.cc)
-test_exe_func(ph_adapt ph_adapt.cc)
+#test_exe_func(ph_adapt ph_adapt.cc)
 test_exe_func(assert_timing assert_timing.cc)
 test_exe_func(create_mis create_mis.cc)
 test_exe_func(fieldReduce fieldReduce.cc)
diff --git a/test/testing.cmake b/test/testing.cmake
index de1120f8..f961ee4a 100644
--- a/test/testing.cmake
+++ b/test/testing.cmake
@@ -69,17 +69,6 @@ else()
   set(GXT dmg)
 endif()
 
-set(MDIR ${MESHES}/phasta/dg)
-
-if(ENABLE_SIMMETRIX AND SIM_PARASOLID AND SIMMODSUITE_SimAdvMeshing_FOUND)
-  set(MDIR ${MESHES}/phasta/BL_query)
-  mpi_test(chef-BL_query 4 ${CMAKE_CURRENT_BINARY_DIR}/chef
-    WORKING_DIRECTORY ${MDIR}/run_case)
-  add_test(NAME chef-BL_query-diff
-    COMMAND diff -r run_case/4-procs_case/ good_case/4-procs_case
-    WORKING_DIRECTORY ${MDIR})
-endif()
-
 if(ENABLE_SIMMETRIX AND SIM_PARASOLID AND SIMMODSUITE_SimAdvMeshing_FOUND)
   if(SIM_DOT_VERSION VERSION_GREATER 12.0.171000)
     set(MDIR ${MESHES}/faceExtrusion)
@@ -670,57 +659,3 @@ if(ENABLE_SIMMETRIX)
       "${MDIR}/crack_nat.x_t")
   endif(SIM_PARASOLID)
 endif()
-if (PCU_COMPRESS)
-  if(ENABLE_SIMMETRIX)
-    set(RUNDIR run_sim)
-  else()
-    set(RUNDIR run)
-  endif()
-  set(MDIR ${MESHES}/phasta/1-1-Chef-Tet-Part/${RUNDIR})
-  if(NOT APPLE)
-    mpi_test(chefStream 1 ${CMAKE_CURRENT_BINARY_DIR}/chefStream
-      WORKING_DIRECTORY ${MDIR})
-  endif()
-  mpi_test(chef0 1 ${CMAKE_CURRENT_BINARY_DIR}/chef
-    WORKING_DIRECTORY ${MDIR})
-  set(MDIR ${MESHES}/phasta/1-1-Chef-Tet-Part)
-  if(ENABLE_SIMMETRIX)
-    add_test(NAME chef1
-      COMMAND diff -r ${RUNDIR}/1-procs_case/ good_phasta/
-      WORKING_DIRECTORY ${MDIR})
-  endif()
-  add_test(NAME chef2
-    COMMAND diff -r out_mesh/ good_mesh/
-    WORKING_DIRECTORY ${MDIR})
-  if(ENABLE_ZOLTAN)
-    mpi_test(chef3 2 ${CMAKE_CURRENT_BINARY_DIR}/chef
-      WORKING_DIRECTORY ${MESHES}/phasta/2-1-Chef-Tet-Part/${RUNDIR})
-    mpi_test(chef4 4 ${CMAKE_CURRENT_BINARY_DIR}/chef
-      WORKING_DIRECTORY ${MESHES}/phasta/2-1-Chef-Tet-Part/4-2-Chef-Part/${RUNDIR})
-    mpi_test(chef5 4 ${CMAKE_CURRENT_BINARY_DIR}/chef
-      WORKING_DIRECTORY ${MESHES}/phasta/4-1-Chef-Tet-Part/${RUNDIR})
-  endif()
-  set(MDIR ${MESHES}/phasta/4-1-Chef-Tet-Part/4-4-Chef-Part-ts20)
-  mpi_test(chef6 4 ${CMAKE_CURRENT_BINARY_DIR}/chef
-    WORKING_DIRECTORY ${MDIR}/${RUNDIR})
-  if(ENABLE_SIMMETRIX)
-    add_test(NAME chef7
-      COMMAND diff -r ${RUNDIR}/4-procs_case/ good_phasta/
-      WORKING_DIRECTORY ${MDIR})
-  endif()
-  add_test(NAME chef8
-    COMMAND diff -r out_mesh/ good_mesh/
-    WORKING_DIRECTORY ${MDIR})
-  if(ENABLE_ZOLTAN AND ENABLE_SIMMETRIX)
-    mpi_test(chef9 2 ${CMAKE_CURRENT_BINARY_DIR}/chef
-      WORKING_DIRECTORY ${MESHES}/phasta/simModelAndAttributes)
-  endif()
-  mpi_test(chefReadUrPrep 4 ${CMAKE_CURRENT_BINARY_DIR}/chefReadUrPrep
-    ../../../model.dmg bz2:../good_mesh/ adapt.ur.inp
-    WORKING_DIRECTORY ${MESHES}/phasta/4-1-Chef-Tet-Part/4-4-Chef-Part-ts20/run)
-  if(ENABLE_ZOLTAN)
-    mpi_test(chefReadRibUrPrep 4 ${CMAKE_CURRENT_BINARY_DIR}/chefReadUrPrep
-      ../../../model.dmg bz2:../good_mesh/ adapt.prerib.inp
-      WORKING_DIRECTORY ${MESHES}/phasta/4-1-Chef-Tet-Part/4-4-Chef-Part-ts20/run)
-  endif()
-endif()
